### PR TITLE
Add SignalFX support to kublet.

### PR DESCRIPTION
**Pull request by jk0**

### CODE REVIEW

File: roles/kube-gateway/templates/gateway.yml.j2 [hunk 1]
```diff
@@ -21,7 +21,7 @@ spec:
     - mountPath: /etc/kubernetes
       name: etc-kubernetes
 {% if datadog_enabled == true %}
-  - image: datadog/docker-dd-agent:latest
```
Comment (bcwaldon) [inline roles/kube-gateway/templates/gateway.yml.j2:24]: Whoops, I missed this in my PR to move onto the `kubernetes` tag. Thank you for finding this

File: roles/kube-gateway/templates/gateway.yml.j2 [hunk 2]
```diff
@@ -32,6 +32,10 @@ spec:
         value: "{{ datadog_api_key }}"
       - name: TAGS
         value: "KubernetesCluster:{{ cluster }},KubernetesClusterRole:{{ cluster_role }}"
+      - name: STATSD_HOST
+        value: {{ datadog_statsd_host|default('') }}
```
Comment (bcwaldon) [inline roles/kube-gateway/templates/gateway.yml.j2]: How about we set the same defaults that are indicated in the config, just in case the underlying code needs to change for any reason in the future

File: roles/kubelet/templates/logspout.yml.j2 [hunk 3]
```diff
@@ -19,7 +19,7 @@ spec:
     - name: SYSLOG_HOSTNAME
       value: '{% raw %}{{ index .Container.Config.Labels "io.kubernetes.pod.namespace" }}{% endraw %}'
     - name: SYSLOG_TAG
-      value: 'klondike-{{ cluster }}-{% raw %}{{ index .Container.Config.Labels "io.kubernetes.pod.namespace" }}-{{ index .Container.Config.Labels "io.kubernetes.pod.name" }}{% endraw %}'
```
Comment (bcwaldon) [inline roles/kubelet/templates/logspout.yml.j2:22]: Why is this change necessary? I very much need logs grouped by pod

Comment (bcwaldon) [reply roles/kubelet/templates/logspout.yml.j2:22]: and I would go one step further and append the container name to the end, but thats a discussion for another review

Comment (jk0) [reply roles/kubelet/templates/logspout.yml.j2:22]: This commit wasn't supposed to make it in here.

**Comment by bcwaldon** [comment]
@jk0 squash all these commits down to one with a sensible commit message, then 👍
